zetaclient/chains/evm/observer/inbound.go

```diff
@@ -192,13 +203,13 @@ func (ob *Observer) ObserveInbound(sampledLogger zerolog.Logger) error {
 	startBlock, toBlock := ob.calcBlockRangeToScan(confirmedBlockNum, lastScanned, config.MaxBlocksPerPeriod)
 
 	// task 1:  query evm chain for zeta sent logs (read at most 100 blocks in one go)
-	lastScannedZetaSent := ob.ObserveZetaSent(startBlock, toBlock)
+	lastScannedZetaSent, _ := ob.ObserveZetaSent(ctx, startBlock, toBlock)
 
 	// task 2: query evm chain for deposited logs (read at most 100 blocks in one go)
-	lastScannedDeposited := ob.ObserveERC20Deposited(startBlock, toBlock)
+	lastScannedDeposited := ob.ObserveERC20Deposited(ctx, startBlock, toBlock)
 
 	// task 3: query the incoming tx to TSS address (read at most 100 blocks in one go)
-	lastScannedTssRecvd := ob.ObserverTSSReceive(startBlock, toBlock)
+	lastScannedTssRecvd, _ := ob.ObserverTSSReceive(ctx, startBlock, toBlock)
```

Comment:
I think we should log the error here?

Reply:
Ideally we need to handle the error if AppContext can't be retrieved (I know it will rarely/never happen).

Reply:
@ws4charlie we indeed handle with in most cases in this PR.